eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/FastaSynonymSequenceReader.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.dbsnp.io;
+
+import uk.ac.ebi.eva.accession.core.io.FastaSequenceReader;
+import uk.ac.ebi.eva.accession.dbsnp.contig.ContigMapping;
+import uk.ac.ebi.eva.accession.dbsnp.contig.ContigSynonyms;
+
+public class FastaSynonymSequenceReader {
```

Comment:
The purpose of this class is to be a `FastaSequenceReader` with some contig mapping additions, in fact the public API is the same and the implementation of some methods the same. In this particular case I think it would be simpler to override the needed couple of methods to achieve that mapping and inherit the rest of implementations.

Reply:
Before the refactor I thought delegation would look better than inheritance, but now I agree that inheritance looks good.